test/new-e2e/tests/remote-config/utils_test.go

```diff
@@ -8,37 +8,91 @@ package remoteconfig
 import (
 	_ "embed"
 	"fmt"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
-	"github.com/stretchr/testify/assert"
+	"os"
+	"path"
+	"strings"
 	"testing"
 	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/stretchr/testify/assert"
 )
 
 // assertLogsEventually will verify that a given `agentName` component's logs contain a pattern.
 // It will continually retry until the `expectedLogPattern` is found or the `maxRetries` is reached,
 // waiting `retryInterval` between each attempt.
 // If the `expectedLogPattern` is not found or an error occurs, the calling test will fail.
-func assertLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogPattern string, waitFor time.Duration, tick time.Duration) {
+func assertAgentLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogs []string, waitFor time.Duration, tick time.Duration) {
 	t.Helper()
+	foundLogs := make(map[string]bool, len(expectedLogs))
+	missingLogs := make([]string, len(expectedLogs))
+	// initially all logs are missing
+	copy(missingLogs, expectedLogs)
+	remoteLogsPath := fmt.Sprintf("/var/log/datadog/%s.log", agentName)
+	t.Logf("looking for logs in %s", remoteLogsPath)
 	assert.EventuallyWithTf(t, func(c *assert.CollectT) {
-		output, err := rh.Execute(fmt.Sprintf("cat /var/log/datadog/%s.log", agentName))
-		if assert.NoError(c, err) {
-			assert.Contains(c, output, expectedLogPattern)
+		// read agent logs
+		agentLogs, err := readRemoteFile(rh, remoteLogsPath)
+		if !assert.NoError(c, err) {
+			return
+		}
+		for _, log := range missingLogs {
+			if strings.Contains(agentLogs, log) {
+				t.Logf("found log: %s", log)
+				foundLogs[log] = true
+			}
 		}
-	}, waitFor, tick, "failed to find log with pattern `%s`", expectedLogPattern)
+		// reset missing logs
+		missingLogs = make([]string, 0, len(expectedLogs))
```

Comment:
Is there a specific reason why the definition of `missingLogs` here is not strictly the same as line 28?

Reply:
Yes, this is a go specificity. When I declare it at line 28 with
```go
make([]string, len(expectedLogs))
```
I create a slice of n empty strings. I use it with `copy` that copies as many items from source as the destination can contain. As `missingLogs` and `expectedLogs` have the same size, `copy` will copy all values from expected
Now here I don't need to copy all values from `expectedLog`, but I expect at most `len(expectedLogs)` items. Instead of creating a slice of size n, I create an empty slice with capacity of n, so that it does not resize at each `append`. 

More details: https://go.dev/tour/moretypes/13